src/components/views/elements/RoomAliasField.tsx

```diff
@@ -91,16 +107,57 @@ export default class RoomAliasField extends React.PureComponent<IProps, IState>
 
     private validationRules = withValidation({
         rules: [
+            { key: "hasSingleDomain",
+                test: async ({ value }) => {
+                    if (!value) {
+                        return true;
+                    }
+                    // Ignore if we have passed domain
+                    if (this.props.domain) {
+                        return true;
+                    }
+                    const split = value.split(':');
+                    if (split.length < 2) {
+                        return false;
+                    }
+                    return true;
+                },
+                invalid: () => _t("Missing domain separator e.g. (:domain.org)"),
+            },
+            {
+                key: "hasMultipleDomains",
+                test: async ({ value }) => {
+                    if (!value) {
+                        return true;
+                    }
+                    // Ignore if we have passed domain
+                    if (this.props.domain) {
+                        return true;
+                    }
+                    const split = value.split(':');
+                    if (split.length > 2) {
+                        return false;
+                    }
```

Comment:
This will break on valid inputs like

`foo:bar.com:8448`
`foo:1080:0:0:0:8:800:200C:417A`

https://spec.matrix.org/v1.1/appendices/#server-name

Reply:
@t3chguy then this check doesn't make sense without proper parser, I'll just throw it out.